packages/fleather/lib/src/widgets/editor_toolbar.dart

```diff
@@ -1289,3 +1300,117 @@
     return widget.child;
   }
 }
+
+const _kMenuScreenPadding = 8.0;
+
+// This is a clone of _PopupMenuRouteLayout from Flutter with some modifications
+class _SelectorLayout extends SingleChildLayoutDelegate {
+  _SelectorLayout(
+    this.position,
+    this.textDirection,
+    this.padding,
+    this.avoidBounds,
+  );
+
+  // Rectangle of underlying button, relative to the overlay's dimensions.
+  final RelativeRect position;
+
+  // Whether to prefer going to the left or to the right.
+  final TextDirection textDirection;
+
+  // The padding of unsafe area.
+  EdgeInsets padding;
+
+  // List of rectangles that we should avoid overlapping. Unusable screen area.
+  final Set<Rect> avoidBounds;
+
+  // We put the child wherever position specifies, so long as it will fit within
+  // the specified parent size padded (inset) by 8. If necessary, we adjust the
+  // child's position so that it fits.
+
+  @override
+  BoxConstraints getConstraintsForChild(BoxConstraints constraints) {
+    // The menu can be at most the size of the overlay minus 8.0 pixels in each
+    // direction.
+    return BoxConstraints.loose(constraints.biggest).deflate(
+      const EdgeInsets.all(_kMenuScreenPadding) + padding,
+    );
+  }
+
+  @override
+  Offset getPositionForChild(Size size, Size childSize) {
+    // size: The size of the overlay.
+    // childSize: The size of the menu, when fully open, as determined by
+    // getConstraintsForChild.
+
+    final double y = position.top;
+
+    // Find the ideal horizontal position.
+    double x;
+    if (position.left > position.right) {
```

Comment:
Can we implement something that depends on TextDirection? 
We want to align to right if enough space in ltr and the other way round in rtl

Reply:
That's what I was thinking at first but thought it might be better to follow what Flutter's doing. But will change it since it's easier.

Reply:
Could you please provide the complete code for me to test the behavior?

Reply:
```dart
@override
Offset getPositionForChild(Size size, Size childSize) {
  // size: The size of the overlay.
  // childSize: The size of the menu, when fully open, as determined by
  // getConstraintsForChild.

  final double y = position.top;

  // Find the ideal horizontal position.
  double x;
  if ((textDirection == TextDirection.ltr ? position.right : position.left) > childSize.width) {
    // Menu button is closer to the left edge, so grow to the right, aligned to the left edge.
    x = position.left;
  } else if ((textDirection == TextDirection.ltr ? position.left : position.right) > childSize.width) {
    // Menu button is closer to the right edge, so grow to the left, aligned to the right edge.
    x = size.width - position.right - childSize.width;
  } else {
    switch (textDirection) {
      case TextDirection.rtl:
        x = size.width - position.right - childSize.width;
      case TextDirection.ltr:
        x = position.left;
    }
  }

  final Offset wantedPosition = Offset(x, y);
  final Offset originCenter = position.toRect(Offset.zero & size).center;
  final Iterable<Rect> subScreens =
      DisplayFeatureSubScreen.subScreensInBounds(
          Offset.zero & size, avoidBounds);
  final Rect subScreen = _closestScreen(subScreens, originCenter);
  return _fitInsideScreen(subScreen, childSize, wantedPosition);
}
```

Reply:
It's not working correctly for RTL case but I get the idea. I don't know which one is better. What do you say? @amantoux

Reply:
In fact, it's not even working correctly for RTL right now. :))

Reply:
Difficult for me to test RTL, how bad is it?

Reply:
I think I fixed it. Thanks for the suggestion! @amantoux